src/modules/feature-modules/innovator/pages/innovation/manage/manage-info.component.ts

```diff
@@ -1,92 +1,102 @@
-
-      for(const innovation of innovationsList.data) {
-        
-        if(innovation.status === InnovationStatusEnum.PAUSED) {
-          this.pausedInnovationsCounter++;
-        }
-
-        if(innovation.status === InnovationStatusEnum.IN_PROGRESS) {
-          this.haveAnyInProgressInnovation = true;
-        }
-
-      }
-
-      this.haveAnyActiveInnovation = innovationsList.data.filter(i => !this.innovationTransfers.map(it => it.innovation.id).includes(i.id))
-        .length
-        > 0;
-
-      this.setPageStatus('READY');
-
-    }
-    );
-  }
-
-
-  cancelInnovationTransfer(transferId: string, innovation: { id: string, name: string }): void {
-
-    this.innovatorService.updateTransferInnovation(transferId, InnovationTransferStatusEnum.CANCELED).pipe(
-      concatMap(() => {
-        this.getInnovationsTransfers();
-        return of(true);
-      })
-    ).subscribe({
-      next: () => {
-        this.setAlertSuccess(`You have cancelled the request to transfer the ownership of '${innovation.name}'`);
-      },
-      error: () => {
-        this.setAlertError('An error occurred when cancelling the transfer. Please try again or contact us for further help');
-      }
-    });
-
-  }
-
-}
+import { Component, OnInit } from '@angular/core';
+import { forkJoin, of } from 'rxjs';
+import { concatMap } from 'rxjs/operators';
+
+import { CoreComponent } from '@app/base';
+
+import { InnovationsService } from '@modules/shared/services/innovations.service';
+import { GetInnovationTransfersDTO, InnovatorService } from '@modules/feature-modules/innovator/services/innovator.service';
+import { InnovationStatusEnum, InnovationTransferStatusEnum } from '@modules/stores/innovation';
+import { ActivatedRoute } from '@angular/router';
+import { ContextInnovationType } from '@modules/stores/context/context.types';
+
+
+@Component({
+  selector: 'app-innovator-pages-innovation-manage-info',
+  templateUrl: './manage-info.component.html'
+})
+export class PageInnovationManageInfoComponent extends CoreComponent implements OnInit {
+
+  innovationId: string;
+  innovation: ContextInnovationType;
+
+  isActiveInnovation = false;
+  isInProgressInnovation = false;
+  innovationTransfers: GetInnovationTransfersDTO = [];
+  pausedInnovationsCounter = 0;
+
+
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private innovationsService: InnovationsService,
+    private innovatorService: InnovatorService
+  ) {
+
+    super();
+    this.setPageTitle('Manage innovation');
+
+    this.innovationId = this.activatedRoute.snapshot.params.innovationId;
+
+    this.innovation = this.stores.context.getInnovation();
+
+    if(this.innovation.status === InnovationStatusEnum.IN_PROGRESS) {
+      this.isInProgressInnovation = true;
+    }
+
+  }
+
+
+  ngOnInit(): void {
+
+    this.getInnovationsTransfers();
+
+  }
+
+  getInnovationsTransfers(): void {
+
+    forkJoin([
+      this.innovationsService.getInnovationsList(),
```

Comment:
If I understood this component is only used for a specific innovation so we shouldn't require to getInnovationsList() and probably we should get transfers for this innovation only instead of all (not sure if the endpoint exists for this already)

Reply:
I kept using the getInnovationsList() because it was used to inform the total number of stopped innovations. If this is not necessary information anymore (I thought it might not be, but I wasn't sure), then we could stop calling getInnovationsList().

I haven't found an endpoint to get only the transfers information about a single innovation, so I kept calling getInnovationTransfers() and filtered the retrieved list